### ISSUE TITLE
no previews generated/shown for local images

### ISSUE THREAD

**Issue by svenb1234** [nextcloud/android#10389]
### Steps to reproduce
1. Let the app upload and move images and videos to the app's folder
2. Start the media view in the app

### Expected behaviour
- previews for all pictures and videos should be generated and shown

### Actual behaviour
- only previews for pictures that are not on the phone are generated and shown. Instead the app shows the grey static image/video icon - not the colour changing icon which is shown while the preview is loaded/generated. Opening the pictures works though.

When swiping, the next local pictures are loaded as well. However, local videos are not loaded when opened via swiping. The display just shows "sorry". Opening the local videos directly from the thumbnail view plays them fine.

### Environment data
Android version: 12

Device model: Pixel 6

Stock or customized system: stock

Nextcloud app version: 3.20.3

Nextcloud server version: 23

**Comment by Tiam-Abderezai** [comment]
@nextcloud-android-bot Please grant access / assign issue to me

**Comment by svenb1234** [comment]
If you do not get a video preview that's probably because ffmpeg or similar is not installed. Also, you are not looking at pictures which alre local, i.e. on your phone's nextcloud folder. At least I do not see any green ticks on your screenshot.

![Screenshot_20220617-071805](https://user-images.githubusercontent.com/46424536/174231769-6e4763e1-7271-4693-94ec-28feac48e6ad.png)

Unfortunately I cannot show you any screenshot of the media view (not file view) since it crashes. But there one would see that those pictures which are only on the server get a preview, the local ones not.

As mentioned above, I use the "move to appp folder after upload feature" of the android app.

**Comment by Tiam-Abderezai** [comment]
@svenb1234  Thanks for the screenshot.
> If you do not get a video preview that's probably because ffmpeg or similar is not installed

Ah that's right, our NC org doesn't have ffmpeg installed. The org I'm a member of isn't hosted by me so I can't add it. 

Therefore I'm going to :
1) install a local NC server and join in as a member
2) install ffmpeg on it
3) test by reproducing the error again by following your steps

>  Also, you are not looking at pictures which are local, i.e. on your phone's nextcloud folder. At least I do not see any green ticks on your screenshot

>  But there one would see that those pictures which are only on the server get a preview, the local ones not. As mentioned above, I use the "move to appp folder after upload feature" of the android app.

Ok, I will take these in consideration as well as soon as I start testing and update you soon with my findings.

**Comment by mlsxlist** [comment]
For me previews in the media overview are also not shown anymore.  Have the impression that this happens since the last Nextcloud android app update.

Once I tap on a symbol the picture or video is loaded correctly.

Huawei P30 pro
Android 10
Nextcloud Android app 3.20.3

![Screenshot_20220618_080413_com nextcloud client_edit_204666903772415](https://user-images.githubusercontent.com/6071818/174425556-e13ff36f-d490-462e-8c7a-ffa0ade52862.jpg)

**Comment by Tiam-Abderezai** [comment]
@AlvaroBrey is **stable-3.20** the correct branch to test for Android version **3.20.3** (the version the user is having issues with) If not, do you know which branch it is so I can test it on?

**Comment by Tiam-Abderezai** [comment]
@AlvaroBrey actually ignore my previous question, found out master branch is currently **3.20.3**

**Comment by Tiam-Abderezai** [comment]
@svenb1234 Question:
When you say 
> However, local videos are not loaded when opened via swiping. The display just shows "sorry"

 does it show this:
![image](https://user-images.githubusercontent.com/6480849/174470355-d6d1498c-9582-4b0b-b1a1-68022d73feb9.png)

> Unfortunately I cannot show you any screenshot of the media view (not file view) since it crashes. 

Can you crash the app like that by going to the media view and provide the logs here?

I tested this on an emulator for Android 10, NextCloud 23, and the current 3.20.3 app version: 

![image](https://user-images.githubusercontent.com/6480849/174470469-13a4ef32-2058-4bba-9cb7-5c02fc254eb9.png)

it shows the image preview there. I'm not sure why yours isn't, but if you provide the crash logs from when you go to media view I can maybe figure out why.

**Comment by Tiam-Abderezai** [comment]
@mlsxlist what is the size of your media folder?

**Comment by mlsxlist** [comment]
@Tiam-Abderezai It has 45.6 GB

**Comment by Tiam-Abderezai** [comment]
@mlsxlist @svenb1234 try clearing the nexcloud app cache on your phones' settings, see if that helps show the previews.

**Comment by mlsxlist** [comment]
@Tiam-Abderezai, this does not help.

**Comment by AlvaroBrey** [comment]
Reproduced. Interestingly enough, when loading the same files in the main activity, the thumbnail is correctly generated, and is displayed correctly in the Media view afterwards.

So, in order to reproduce:
1. Select a bunch of pictures to upload
2. Select "move to nextcloud folder" as upload behaviour
3. Immediately go into media view

**Comment by AlvaroBrey** [comment]
@mlsxlist your problem looks different from the original issue. Please open a separate issue.

**Comment by svenb1234** [comment]
What do you mean by "main view" for me switching to the normal "all files" view and returning to the media view does not help.

I have a folder in my user with the photos in it. That folder is shared to a second user. If I open the android app, all photos that are on my phone as well as on the server are not shown with previews. The second user's pictures are shown with a preview.

If I open the media view on the second user's phone, it is the same, just the other way around. The pictures on the second user's phone are not shown, while mine are shown fine.

In the web-ui, all pictures are shown with preview.

I do remember though that I did have situations where images were shown without preview and after changing to "all files" and back, they were shown. But currently it is as described. Previews are never shown for pictures which are also in the local device's nextcloud app folder.

Maybe unrelated: in the media view the green tick for local pictures is missing. That tick does show-up in the file view of the folder they are in.

Also, the peviews are not shown/generated in the file view either. But that might also be related to the folder having a lot of files and the file view only showing 155 of them, even when refreshing. LArge folders have been an issue for the app for a long time. I cannot find anything in the logs though only the double "//" irritates me a bit.

`78.49.52.111 - - [22/Jun/2022:16:27:11 +0000] "PROPFIND /remote.php/dav/files/user//Familienfotos/user%20Telefon/Camera/ HTTP/1.1" 207 100763 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.20.3"`

**Comment by svenb1234** [comment]
If I use a third device, a tablet which belongs to the first user and has no pictures on it, it generates and shows all previews of user one and two. Generating again because they are larger?

In any case, after that third device which has no local pictures on it has opened the media view, the previews are also shown on the device of the first user with local pictures. The second user's device still does not show the previews for the pictures on the second user's device.

However, the first few times one opens the media view on the first user's device with the local pictures on it, one gets crashes of the attached type. After some crashes the it seems that whatever causes them is solved and everything works without these crashes. Until there are new pictures and the process begins all over. Some cache which is built?

### Cause of error
```java
Exception in thread "main" java.lang.IllegalStateException: Couldn't read row 709, col 3 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
    at android.database.CursorWindow.nativeGetString(Native Method)
    at android.database.CursorWindow.getString(CursorWindow.java:462)
    at android.database.AbstractWindowedCursor.getString(AbstractWindowedCursor.java:54)
    at android.database.CursorWrapper.getString(CursorWrapper.java:141)
    at com.owncloud.android.datamodel.FileDataStorageManager.createFileInstance(FileDataStorageManager.java:957)
    at com.owncloud.android.datamodel.FileDataStorageManager.getGalleryItems(FileDataStorageManager.java:2224)
    at com.owncloud.android.datamodel.FileDataStorageManager.getAllGalleryItems(FileDataStorageManager.java:2169)
    at com.owncloud.android.ui.adapter.GalleryAdapter.showAllGalleryItems(GalleryAdapter.kt:149)
    at com.owncloud.android.ui.fragment.GalleryFragment.showAllGalleryItems(GalleryFragment.java:262)
    at com.owncloud.android.ui.asynctasks.GallerySearchTask.onPostExecute(GallerySearchTask.java:127)
    at com.owncloud.android.ui.asynctasks.GallerySearchTask.onPostExecute(GallerySearchTask.java:48)
    at android.os.AsyncTask.finish(AsyncTask.java:771)
    at android.os.AsyncTask.access$900(AsyncTask.java:199)
    at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:788)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:7870)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30200390`
* Build flavor: `gplay`

### Device information
* Brand: `google`
* Device: `oriole`
* Model: `Pixel 6`
* Id: `SQ3A.220605.009.B1`
* Product: `oriole`

### Firmware
* SDK: `32`
* Release: `12`
* Incremental: `8650216`

**Comment by AlvaroBrey** [comment]
> In any case, after that third device which has no local pictures on it has opened the media view, the previews are also shown on the device of the first user with local pictures. The second user's device still does not show the previews for the pictures on the second user's device.

Wow, this is very unexpected. Thanks for the detailed investigation.

> However, the first few times one opens the media view on the first user's device with the local pictures on it, one gets crashes of the attached type. 

This is the same crash detailed in #10213, can you reply there?

**Comment by svenb1234** [comment]
> does it show this:
> ![image](https://user-images.githubusercontent.com/6480849/174470355-d6d1498c-9582-4b0b-b1a1-68022d73feb9.png)

Yes, this is what I see when swiping through the pictures as soon as a locally stored video is "opened".

**Comment by svenb1234** [comment]
> In any case, after that third device which has no local pictures on it has opened the media view, the previews are also shown on the device of the first user with local pictures. The second user's device still does not show the previews for the pictures on the second user's device.

I cannot reproduce this anymore. Not sure why it worked before. 

So I am back to: local pictures are not shown as preview.

**Comment by pracovnik** [comment]
Sony XA2 ... fdroid instance of nextcloud android application (newest). Issue is long-term with numerous updates + firmware refresh/clean.downgrade. I do not think this is related to server version.

Issue is the thumbnail in file picker as it doesn't get updated

(1)
I make a 5 photos of triangle. Photo get name DSC001.jpg to DSC005.jpg .
Via nextcloud android application ... I go to "all-files" in which in some (nexcloud) folder I click "upload files"  with "delete after" option. So I go to picture directory on phone and see pictures, where I choose those 5 pictures DSC001.jpg (thumbnail is triangle on each) to upload. It get properly uploaded to nextcloud as I would expect.

(2)
I make 6 new photos of a cube. Photo get name DSC001.jpg - DSC006.jpg , as file with that name don't exist in DCIM folder.
Via nextcloud android application ... I go to a nextcloud folder(in application, different as above) and click "upload files"  with "delete after" option. So I go to picture directory on phone and I see DSC001.jpg-DSC005.jpg with triangle thumbnail and DSC006.jpg with cube as thumbnail. I click upload, it get properly uploaded to nextcloud as I would expect. And the photos DSC001.jpg - DSC006.jpg show cube.

**Comment by svenb1234** [comment]
https://help.nextcloud.com/t/android-app-no-thumbnails-in-media-view/139773

Forum thread with users that share the same issue.

**Comment by SofianeLasri** [comment]
The problem still occurs. Some pictures are showing up, others not.
[logcatOutput.txt](https://github.com/nextcloud/android/files/9828896/logcatOutput.txt)
![Screenshot_20221020-123657_Nextcloud](https://user-images.githubusercontent.com/43948827/196927043-d5d2d72c-5997-41e9-b408-2e48ae81e9e7.jpg)
![Screenshot_20221020-123641_Nextcloud](https://user-images.githubusercontent.com/43948827/196927049-5e5772ea-90ef-4046-a24d-e2b7c21d114a.jpg)
![Screenshot_20221020-123650_Nextcloud](https://user-images.githubusercontent.com/43948827/196927050-aced4b39-741d-41fe-a4c3-bea314c21447.jpg)

**Comment by SofianeLasri** [comment]
The bug persists in One UI 5 (android 13). Only low resolutions pictures are shown.

**Comment by photomatix18** [comment]
I too have this issue.

**Comment by svenb1234** [comment]
> A reinstall of nextcloud app seems to fix it and previews are requested.

Does it fix it permanently?

**Comment by SofianeLasri** [comment]
> > A reinstall of nextcloud app seems to fix it and previews are requested.
> 
> Does it fix it permanently?

It seems working fine.

**Comment by svenb1234** [comment]
@AlvaroBrey so how is this "fix" coming to the many users that do have the app already installed and do not reinstall apps on a regular basis?

**Comment by SofianeLasri** [comment]
No in fact it only temporarily fixed the problem. It now occurs again.

**Comment by AlvaroBrey** [comment]
> @AlvaroBrey so how is this "fix" coming to the many users that do have the app already installed and do not reinstall apps on a regular basis?

I was able to reproduce this problem (or another with similar symptoms), but due to other priorities I am unable to work on debugging it. Unfortunately both the media view code AND the thumbnail generation code are a bit of a mess, so it's going to take some time to work through it, and I don't have that time right now.

**Comment by work-flower** [comment]
Same behaviour on fairphone 4. Just to have a note on it...

**Comment by PrzemekSkw** [comment]
Hello, I reinstall app and see all photos but after I make new I don't see them. So it's not normal to reinstall app all the time I make new photos with smartphone 😕
Any news when app will be fixed?

**Comment by pbfreak** [comment]
Hello, same Problem here. Preview of Photos are not Shows. After i switch to "Files" and reload a folder with Pictures i have preview pictures of this Photos in Files and Media

**Comment by chrissi5120** [comment]
#11431 pretty similar behavior for me, is anyone working on a solution?

**Comment by AxTech212** [comment]
I just want to metion - i have the same problem - only with androiddevices and i have the problem since more than one year - there seems to be a bigger problem.

**Comment by chrissi5120** [comment]
> I just want to metion - i have the same problem - only with androiddevices and i have the problem since more than one year - there seems to be a bigger problem.

Yeah, this is going on “forever” on my side, sorry for this gloss-over description… I got so used to switch back to the folders and check media tab again, that I could not name a point in time when this all started.

One weird thing is, that it lately seems to work fine 90% of the time on my wives Galaxy Tab S7. This is my only tablet-device and the only difference I see immediately, is that there is the bigger screen and the propable different preview being used here?

All other android devices "in service" for my family fail, right now Pixel 6a, Galaxy S8, Galaxy A 53, Moto G 12-something..

This is the number on topic of discussion in my family regarding this app.

**Comment by MartinFialka** [comment]
> > @AlvaroBrey so how is this "fix" coming to the many users that do have the app already installed and do not reinstall apps on a regular basis?
> 
> I was able to reproduce this problem (or another with similar symptoms), but due to other priorities I am unable to work on debugging it. Unfortunately both the media view code AND the thumbnail generation code are a bit of a mess, so it's going to take some time to work through it, and I don't have that time right now.
It would be very nice to clean that up. I'm also having the same issue on android 13, pixel 7.

**Comment by chrissi5120** [comment]
Hey @AlvaroBrey any chance you could look into that? Issue is persisting for all my Android devices.

**Comment by fuzunspm** [comment]
I have the same issue. Previews are working on web browser but on android app they are missing. Clicking on an image correctly displays full quality image

**Comment by francescotili** [comment]
So it's now June 2024, the issue is still open. Are there any news on this??

**Comment by pbfreak** [comment]
Still having the Problem. Switched to Photopris Works perfectly. If you want to only sync Photos/Videos have a look

**Comment by SofianeLasri** [comment]
> Still having the Problem. Switched to Photopris Works perfectly. If you want to only sync Photos/Videos have a look

I think you can still wait several months, they prefer to focus on adding futile AI-based features, rather than fixing bugs.

**Comment by SofianeLasri** [comment]
> So it's now June 2024, the issue is still open. Are there any news on this??

I think you can still wait several months, they prefer to focus on adding futile AI-based features, rather than fixing bugs.

**Comment by chrissi5120** [comment]
My workflow for Android: 

Don’t use the gallery, it’s useless. I gave up. 

IOS: 

all fine, except stuff is not uploading in background and I have to ask people to keep the app open for that to happen. Probably apples fault.

Desktop/Laptop/Tablet: 

just use memories, it is better in every way.

I would like to contribute good money specifically for that feature to work normally, it is that big of a pain in the…

**Comment by SofianeLasri** [comment]
If it were possible, I would not use Nextcloud at all. The problem is that it is literally the only self-hostable solution that is mature enough to store all my files while being able to access them from all my devices. If there was another similar paid solution, I would have used it with pleasure.

I've been using Nextcloud for several years (2016, at the time it was Owncloud), and I still can't convince my family and friends to use it despite the fact that I manage the server, and therefore the storage space. It must be said that the platform is not a dream, quite the contrary! The web interface still looks cheap (I even have the impression that it gets worse year after year), the mobile application is still full of bugs, and the desktop application seems to come straight from the early 2000s.
They much prefer to hurt themselves by trying not to exceed the storage space provided for free by Google, Apple or Microsoft, rather than using this "abomination" that Nextcloud represents for them.

And the worst part is that with each major update, Nextcloud doesn't seem to fix many bugs, but on the contrary, seems to add a ton of features that are almost useless.

"You didn't ask for face recognition? Well, we added one, and it works very badly! :D
You didn't ask for a totally useless integration with the famous "Artificial Intelligence"? Well, we added one too! But since that wasn't enough, it's up to you to create the applications to make it work. And yes, we couldn't be bothered to do it (it would have been too bad anyway, so we might as well do nothing :) )".

**Comment by camillo-positano** [comment]
Hello i've had this issue for a while as well. 
i created a pull request (#13158 ) that will fix this issue. Got a first approval and waiting on a merge!

Hope it fixes it for everybody

**Comment by francescotili** [comment]
> i created a pull request (#13158 ) that will fix this issue.

OMG yes thank you 😊

**Comment by camillo-positano** [comment]
It has been merged, we'll see if it fixes this issue when it gets released

**Comment by marvinnitz18** [comment]
3.30 release for android, when ❓

**Comment by francescotili** [comment]
> 3.30 release for android, when ❓

I am interested too, the update release speed is extremely slow, is this normal? It's been two months and it's still not in production?

**Comment by camillo-positano** [comment]
I dont think before september/october because august (at least in italy) is vacation month.
Dont know if its in all europe tho

**Comment by joshtrichards** [comment]
Fixed in #13158
